session_ii/javascript-fundamentals/session1/homework/raluca/classes.js

```diff
@@ -0,0 +1,75 @@
+// 1. write a base class called RegularShape whose constructor receives the number
+//    of sides and the length of a side. write at least two derived classes that extend
+//    RegularShape, representing two different regular shapes,
+//    containing a method called getArea, which must return the area of the
+//    shape, using the two arguments that the parent class should receive.
+
+// 1. Write a base class called RegularShape whose constructor receives the number of sides and the length of a side. 
+// Write at least two derived classes that extend RegularShape, representing two different regular shapes, containing a method called getArea, which must return the area of the shape, using the two arguments that the parent class should receive.
+
+class RegularShape {
+	constructor(numberOfSides, sideLength, shapeName) {
+		this.numberOfSides = numberOfSides;
+		this.sideLength = sideLength;
+		this.shapeName = shapeName;
+	}
+
+	calculateArea() {
+		const cot = 1 / Math.tan(Math.PI / this.numberOfSides)
+		const area = this.numberOfSides * Math.pow(this.sideLength, 2) * cot / 4;
+
+		console.log(`${this.shapeName} Area:`, area.toFixed(2))
+	}
+}
+
+class Triangle extends RegularShape {
+	constructor(numberOfSides, sideLength, shapeName) {
+		super(numberOfSides, sideLength, shapeName)
+	}
+
+	getArea() {
+		super.calculateArea()
+	}
+}
+
+class Square extends RegularShape {
+	constructor(numberOfSides, sideLength, shapeName) {
+		super(numberOfSides, sideLength, shapeName);
+	}
+
+	getArea() {
+		super.calculateArea()
+	}
+}
+
+class Pentagon extends RegularShape {
+	constructor(numberOfSides, sideLength, shapeName) {
+		super(numberOfSides, sideLength, shapeName)
+	}
+
+	getArea() {
+		super.calculateArea()
+	}
+}
+
+class Hexagon extends RegularShape {
+	constructor(numberOfSides, sideLength, shapeName) {
+		super(numberOfSides, sideLength, shapeName);
+	}
+
+	getArea() {
+		super.calculateArea()
```

Comment:
the `getArea` methods should return the result (for instance we might want to print the result to the console). Of course, since you implemented the general formula, you could rename the inherited method to 'getArea' and use that instead of defining it in all the sublcasses. :)